File: Algorithmique appliquée/chapitre2/exercice1.cpp
/* programme polynome
*  but : saisie d'un polynome et calcul de f(x) en fonction de x
   date: 05/10/2016
*/
#include <iostream>
#include <math.h>
using namespace std;

int main(int argc, char *argv[]){
    unsigned degre;
    int k;
    int a[6];
    float resultat, x;
    char choix;

    do{
        cout << "Entrez le degré du polynome (maximum 6) =";
        cin >> degre;
    }while(degre >6);

    //saisie des termes du polynômes
    for(k=0; k <=degre;k++){
        cout << "a" << k << "=";
        cin >> a[k];
    }

    //affichage du polynome
    cout << "f(X) =" << a[0];
    for(k=1; k<=degre;k++){
        cout << "+" << a[k] << "*X^" << k;
    }
    cout << endl;

    do{
        cout << "X =";
        cin >> x;
        resultat =0;
        for(k=0; k<=degre;k++){
            resultat += a[k] * pow(x,k);
        }
        cout << "f(X) = " << resultat << "\n";
        // demande si l'utilisateur veut continuer
        cout << "Voulez-vous continuer?(O/N) =";
        cin >> choix;

    }while(choix == '0' || choix == 'o');

    return 0;

}

File: Algorithmique appliquée/chapitre2/python/exercice1.py
# -*-coding:utf-8 -*
# Programme polynome
# saisie d'un polynome et calcul de f(x) en fonction de x
# date : 05/10/2016


import os
import math


# saisie du polynôme obligatoirement <= 6
degre = 7
while degre > 6 :
    degre = int(input("Entrez le degré du polynome (maximum 6) = "))

# déclaration du tableau
a = [0.0]*(degre+1)

# saisie des termes du polynome
for k in range(0, degre+1) :
    a[k] = float(input("a" + str(k) + " = "))

# affichage du polynôme
print("\nf(X) = ", a[0])
for k in range(1, degre+1) :
    print(" + ", a[k], "*X^", k)
print("\n")

# boucle sur la saisie des valeurs de x et le calcul de f(x)
choix = "O"
while choix == "O" or choix == "o" :
    # saisie de x
    x = float(input("X = "))
    # calcul de f(x)
    resultat = 0.0
    for k in range(0, degre+1) :
        resultat += a[k] * math.pow(x, k)
    print("f(X) = ", resultat)
    # demade si l'utilisateur veut continuer
    choix = input("\nVoulez-vous continuer ? (O/N) = ")


os.system("pause")
